src/migrations.rs

```diff
@@ -219,6 +261,149 @@ impl Contract {
+    pub label_to_posts: UnorderedMap<String, HashSet<PostId>>,
+    pub access_control: AccessControl,
+    pub authors: UnorderedMap<AccountId, HashSet<PostId>>,
+    pub communities: UnorderedMap<String, CommunityV2>,
+    pub featured_communities: Vec<FeaturedCommunity>,
+}
+
+// From ContractV6 to ContractV7
+#[near_bindgen]
+impl Contract {
+    fn unsafe_add_projects() {
+        let ContractV6 {
+            posts,
+            post_to_parent,
+            post_to_children,
+            label_to_posts,
+            access_control,
+            authors,
+            mut communities,
+            featured_communities,
+        } = env::state_read().unwrap();
+
+        let migrated_communities: Vec<(String, CommunityV3)> = communities
+            .iter()
+            .map(|(community_handle, community)| {
+                (
+                    community_handle,
+                    CommunityV3 {
+                        handle: community.handle,
+                        admins: community.admins,
+                        name: community.name,
+                        description: community.description,
+                        bio_markdown: community.bio_markdown,
+                        logo_url: community.logo_url,
+                        banner_url: community.banner_url,
+                        tag: community.tag,
+                        github_handle: community.github_handle,
+                        telegram_handle: community.telegram_handle,
+                        twitter_handle: community.twitter_handle,
+                        website_url: community.website_url,
+                        github: community.github,
+                        wiki1: community.wiki1,
+                        wiki2: community.wiki2,
+                        project_ids: Default::default(),
+
+                        feature_flags: CommunityFeatureFlags {
+                            github_integration: true,
+                            projects: true,
+                            sponsorship: true,
+                            wiki: true,
+                        },
+                    },
+                )
+            })
+            .collect();
+
+        communities.clear();
+
+        let mut communities_new = UnorderedMap::new(StorageKey::Communities);
+
+        for (k, v) in migrated_communities {
+            communities_new.insert(&k, &v);
+        }
+
+        env::state_write(&ContractV7 {
+            posts,
+            post_to_parent,
+            post_to_children,
+            label_to_posts,
+            access_control,
+            authors,
+            communities: communities_new,
+            featured_communities,
+            last_project_id: 0,
+            projects: UnorderedMap::new(StorageKey::Projects),
+            project_views: UnorderedMap::new(StorageKey::Projects),
+        });
+    }
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Clone)]
+pub struct CommunityV3 {
```

Comment:
What is the reason to introduce community v3? Instead of import struct community

Reply:
@ailisp to keep migrations in boundaries of "history" convention, because we're potentially going to have V4, V5, etc.

( CC @frol )